preprocessing/save_v_mmap.py

```diff
@@ -0,0 +1,46 @@
+import pandas as pd
+import numpy as np
+import matplotlib.pyplot as plt
+import pathlib
+
+data_path = pathlib.Path("datasets")
+data_path.mkdir(exist_ok=True)
+
+df = pd.read_csv(
+    "1958-perceptron/iris.data", 
+    header=None,
+    names=[
+        "sepal_length", 
+        "sepal_width", 
+        "petal_length", 
+        "petal_width", 
+        "class"
+    ],
+    dtype={
+        "sepal_length": np.float32, 
+        "sepal_width": np.float32, 
+        "petal_length": np.float32, 
+        "petal_width": np.float32, 
+        "class": "category",
+    },
+    engine="pyarrow"
+    )
+
+xs, ys = (
+    df.loc[:, df.columns != "class"].to_numpy(), 
+    df["class"].cat.codes.to_numpy()
+)
```

Comment:
not sure if this category type is actually efficient, but it's a whole lot simpler than "fitting" a label encoder with sklearn or rolling your own logic to apply. 

this lets you pull out either `cat.codes` or `cat.categories` - although if you were doing a buffered read, you would probably need to [pre-define these columns](https://pandas.pydata.org/docs/user_guide/categorical.html)